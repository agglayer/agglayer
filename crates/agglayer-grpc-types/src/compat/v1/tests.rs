use agglayer_types::primitives::SignatureError;

use super::Error;

#[rstest::rstest]
#[case::error("no_proof", Error::missing_field("proof"))]
#[case::error("bad_data", Error::invalid_data("invalid value".to_owned()))]
#[case::error("bad_data_in_field", Error::invalid_data("invalid value".to_owned()).inside_field("value"))]
#[case::error("bad_data_in_nested", Error::invalid_data("invalid value".to_owned()).inside_field("value").inside_field("data"))]
#[case::error("failed_ser", Error::serializing_proof(Box::new(bincode::ErrorKind::Custom("failed".to_owned()))))]
#[case::error("failed_deser", Error::deserializing_proof(Box::new(bincode::ErrorKind::Custom("failed".to_owned()))))]
#[case::error("bad_sig", Error::parsing_signature(SignatureError::InvalidParity(5)))]
#[case::error("bad_sig_in_nested", Error::parsing_signature(SignatureError::InvalidParity(5)).inside_field("signature").inside_field("data"))]
fn error_messages(#[case] name: &str, #[case] error: Error) {
    insta::assert_snapshot!(format!("{name}/display"), error);
    insta::assert_debug_snapshot!(format!("{name}/kind"), error.kind());
    insta::assert_snapshot!(
        format!("{name}/debug"),
        format!("{:?}", anyhow::Error::from(error))
    );
}
