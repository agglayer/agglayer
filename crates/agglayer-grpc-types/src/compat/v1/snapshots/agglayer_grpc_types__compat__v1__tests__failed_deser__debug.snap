---
source: crates/agglayer-grpc-types/src/compat/v1/tests.rs
expression: "format!(\"{:?}\", anyhow::Error::from(error))"
---
failed to deserialize proof

Caused by:
    failed
